rest-services/server/src/main/java/com/_4point/aem/docservices/rest_services/server/assembler/AssembleDocuments.java

```diff
@@ -0,0 +1,209 @@
+
+@SuppressWarnings("serial")
+@Component(service = Servlet.class, property = {
+		Constants.SERVICE_DESCRIPTION + "=FormsService.AssembleDocuments Service",
+		"sling.servlet.methods=" + HttpConstants.METHOD_POST })
+@SlingServletPaths("/services/AssemblerService/AssembleDocuments")
+public class AssembleDocuments extends SlingAllMethodsServlet {
+
+	private static final Logger log = LoggerFactory.getLogger(AssembleDocuments.class);
+
+	@Reference
+	private com.adobe.fd.assembler.service.AssemblerService adobeAssembleService;
+
+	private final Supplier<TraditionalDocAssemblerService> assemblerServiceFactory = this::getAdobeAssemblerService;
+	
+
+	private final DocumentFactory docFactory = DocumentFactory.getDefault();
+
+	@Override
+	protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response)
+			throws ServletException, IOException {
+		try {
+			this.processInput(request, response);
+		} catch (BadRequestException br) {
+			log.warn("Bad Request from the user", br);
+			response.sendError(SlingHttpServletResponse.SC_BAD_REQUEST, br.getMessage());
+		} catch (InternalServerErrorException ise) {
+			log.error("Internal server error", ise);
+			response.sendError(SlingHttpServletResponse.SC_INTERNAL_SERVER_ERROR, ise.getMessage());
+		} catch (NotAcceptableException nae) {
+			log.error("NotAcceptable error", nae);
+			response.sendError(SlingHttpServletResponse.SC_NOT_ACCEPTABLE, nae.getMessage());
+		} catch (Exception e) { // Some exception we haven't anticipated.
+			log.error(e.getMessage() != null ? e.getMessage() : e.getClass().getName(), e); // Make sure this gets into
+																							// our log.
+			throw e;
+		}
+	}
+
+	private void processInput(SlingHttpServletRequest request, SlingHttpServletResponse response)
+			throws BadRequestException, InternalServerErrorException, NotAcceptableException {
+		// TODO Auto-generated method stub
+		AssemblerService assemblerService = new AssemblerServiceImpl(assemblerServiceFactory.get(),
+				UsageContext.SERVER_SIDE);
+		log.debug("In Assemble Uploaded Files");
+		try {
+			Map<String, Object> mapOfDocuments = new HashMap<String, Object>();
+			InputStream dDXFile = createDdxFile(request, mapOfDocuments);
+			Document ddx = docFactory.create(dDXFile);
+			try (Document result = executeOn(ddx, mapOfDocuments, assemblerService)) {
+				String contentType = result.getContentType();
+				ServletUtils.validateAcceptHeader(request.getHeader(AcceptHeaders.ACCEPT_HEADER_STR), contentType);
+				response.setContentType(contentType);
+				response.setContentLength((int) result.length());
+				ServletUtils.transfer(result.getInputStream(), response.getOutputStream());
+			}
+		} catch (AssemblerServiceException | IOException e) {
+			// TODO Auto-generated catch block
+			throw new InternalServerErrorException("Internal Error while merging PDF. (" + e.getMessage() + ").", e);
+		} catch (Exception e) {
+			log.error("Error while Merging pdfs " + e.getMessage());
```

Comment:
Catching Exception outside of the top level routine is almost always wrong as is catching an exception, logging it and then carrying on.

In general, the error handling strategy applied in the fluent forms library is to catch an exception, provide additional context to it by wrapping it and rethrowing the exception so that the exception eventually bubbles up to the top level.  Also, any external exceptions are typically wrapped with one of our own exceptions unless it is one that the user might wish to otherwise handle themselves (such as FileNotFoundException).

This catch(Exception e) block should probably be removed.